### PR TITLE
Adding prettier and eslint to frontend

### LINKED ISSUE #588: Add Prettier and ESLint to Frontend Develop Branch
### Overview
As a developer, we need to ensure our code adheres to quality standards and remains maintainable. For this issue, we will configure Prettier and ESLint for the frontend to enforce consistent code formatting and identify any linting errors. This setup will help us keep our code clean and improve overall code quality.

### Action Items
- [x] Install Prettier for Dev env 
- [x] Install ESLint and configure it
- [x] Create a script for both in package.json

### Resources/Instructions
https://eslint.org/docs/latest/use/configure/
https://blog.logrocket.com/linting-typescript-eslint-prettier/

### PR DESCRIPTION
**Pull request by irais-valenzuela**
Fixes #588 

### Changes
- In package.json under devDependencies I installed prettier and EsLint and wrote scripts to run them. 
- Created a eslint.config.mjs file to configure ESLint.

**Side Notes:** 
- To the linter rules config I just added basic rules for now like indentation, no whitespaces, etc.. we can discuss others we'd like to add. 
- I also tested the linter with poorly written ts code and tsx code and the linter catches the errors and prettier fixes them.

### CODE REVIEW

**Comment by LoTerence** [comment]
Please change the base branch to `hackforla:develop`

**Comment by LoTerence** [comment]
Running `eslint .` works for me -- so I can confirm eslint is set up correctly. Nice!

Also checked if it detects typescript errors and it works - good job 👍 

Adding a formatter and a linter to the frontend will be very important to long term health of our project. So thank you for doing this.

There are some problems I would like you to fix

1. Running the `npm run lint` command doesnt work for me:
image

Can you fix that script please? I think `eslint .` should be enough

2. There's 471 issues found by the linter:

image

If we merge this into `develop`, everyone will have a lot of linter errors next time they pull from the develop branch.

Could you try running `eslint . --fix` to automatically resolve as many of them as possible? Lets keep it in this PR.

We can inform the team of the new changes and make an issue to resolve the rest of them after.

**Comment by irais-valenzuela** [comment]
Update: I ran` ESLint . --fix `to automatically resolve most linting errors. The remaining issues include console logs, some unused variables, type errors declared as "any," and prop validation concerns. I went through and fixed some of these manually as well. While these issues won't impact our project at the moment, they should be addressed in the future. @LoTerence What are your thoughts on whether we should fix the "any" type errors and prop validation now, or can we prioritize them for later?

Additionally, I've added the TailwindCSS plugin with the following rules:

```
"tailwindcss/no-contradicting-classname": "error",
"tailwindcss/no-unnecessary-arbitrary-value": "error",
```
I tested the configuration, and the linter successfully caught the relevant errors.

**Comment by LoTerence** [comment]
Thank you for the updated PR! Great job on fixing the npm script and running the autofix 💯 Thank you for taking the time to manually fix others as well! The tailwind plugins look good too 👌

> What are your thoughts on whether we should fix the "any" type errors and prop validation now, or can we prioritize them for later?

Yes, lets prioritize them later. Could you make an issue with the details you found about the remaining linter errors? (Or add it to the already existing issue, I will leave it up to you).

We will get the whole frontend team to help with this. It will be good practice for you all.

File: frontend/src/pages/CreditsPage/CreditsPage.tsx
Comment (LoTerence) [inline frontend/src/pages/CreditsPage/CreditsPage.tsx:1]: Do we still need this line if we are using the rule `"react/no-unescaped-entities": "off",` ?

I didn't test it so I dont know 😕

If we don't need this line anymore could you remove it from all files you added it to?

File: frontend/src/pages/NotFoundPage/NotFoundPage.tsx
Comment (LoTerence) [inline frontend/src/pages/NotFoundPage/NotFoundPage.tsx:1]: like this one, and I think I saw a couple more

**Review by LoTerence — CHANGES_REQUESTED** [review]
After these small changes it should be ready to merge 👍

File: frontend/src/pages/NotFoundPage/NotFoundPage.tsx
Comment (irais-valenzuela) [reply frontend/src/pages/NotFoundPage/NotFoundPage.tsx:1]: Hey @LoTerence, I'm prepping for an interview so I'll get to this probably this weekend.

Comment (irais-valenzuela) [reply frontend/src/pages/NotFoundPage/NotFoundPage.tsx:1]: Hey @LoTerence, I'm prepping for an interview so I'll get to this probably this weekend.

Comment (LoTerence) [reply frontend/src/pages/NotFoundPage/NotFoundPage.tsx:1]: No problem, good luck on your interview!

Comment (irais-valenzuela) [reply frontend/src/pages/NotFoundPage/NotFoundPage.tsx:1]: Hey @LoTerence, apologies for being MIA—I’ve been caught up with interviews. I added the plugins for hooks, accessibility, and the Tailwind classnames-order rule, and tested them all, so they’re working well. For the /* eslint-disable react/no-unescaped-entities */ question, I'm still investigating why the global rule in the ESLint config isn't taking effect. For now, adding the comment in the file bypasses the error successfully.

Since I’ve spent a bit of time on this, would you like me to close this as-is, or would you prefer I keep investigating a solution in the config? Let me know your thoughts!

Comment (LoTerence) [reply frontend/src/pages/NotFoundPage/NotFoundPage.tsx:1]: Great job thanks!

> Since I’ve spent a bit of time on this, would you like me to close this as-is, or would you prefer I keep investigating a solution in the config?
Ok thank you for investigating! No problem, we'll keep the comments there for now, and clean them up when we resolve the rest of the linter errors later.

> I also wrote some docs on the configurations and rules I added to the linter file.
Awesome! Thank you for doing this! I will review now.

File: docs/developer/frontend.md
Comment (LoTerence) [inline docs/developer/frontend.md:228]: Beautiful documentation! Love it! Thanks for doing this!

**Review by LoTerence — APPROVED** [review]
Everything looks great! Great job Ida!

We just need to resolve the merge conflicts now, and it will be ready to merge :) Would you mind doing that really quickly?

**Comment by irais-valenzuela** [comment]
Hey @LoTerence, it seems I don't have write access to fix the merge conflicts. Is this something I need to do on my end to give myself access?

**Comment by LoTerence** [comment]
> Hey @LoTerence, it seems I don't have write access to fix the merge conflicts. Is this something I need to do on my end to give myself access?

Its not something you need access to on github, you have to pull the newest changes from `hackforla:develop` into your branch, make changes to resolve the merge conflicts, then push them back up to github.

Its a lot to explain, check out this video to see what I'm talking about:
https://www.youtube.com/watch?v=xNVM5UxlFSA

**Comment by irais-valenzuela** [comment]
@LoTerence, I fixed the merged conflicts. Should be able to merge the PR now :)

**Comment by LoTerence** [comment]
Awesome! Amazing job! Go ahead and click the merge button whenever you are ready 😄

**Comment by irais-valenzuela** [comment]
@LoTerence, thanks! I don't think I can merge it says "You’re not to merge this pull request." Is there something I do on my end?

**Comment by LoTerence** [comment]
Sorry about that, just fixed the project settings so you should be able to click the merge button now 👌

**Pull request merged**
